cassandra/cqlengine/models.py

```diff
@@ -357,26 +357,26 @@ class MultipleObjectsReturned(_MultipleObjectsReturned):
     _table_name = None  # used internally to cache a derived table name
 
     def __init__(self, **values):
-        self._values = {}
         self._ttl = self.__default_ttl__
         self._timestamp = None
         self._transaction = None
+        self._batch = None
+        self._timeout = connection.NOT_SET
+
+        # A flag set by the deserializer to indicate that update should be used
+        # when persisting changes.
+        self._is_persisted = values.pop('_is_persisted', False)
 
+        # Initialize each volumn to its provided or default right at
+        # instanciation. See PYTHON-348.
+        self._values = {}
         for name, column in self._columns.items():
             value = values.get(name, None)
             if value is not None or isinstance(column, columns.BaseContainerColumn):
                 value = column.to_python(value)
             value_mngr = column.value_manager(self, column, value)
-            if name in values:
-                value_mngr.explicit = True
```

Comment:
Yes, I also worry about all the None/empty/null edge cases but I was pretty sure I didn't messed that up. Anyway, #433 seems to address this very issue.